### PR TITLE
DF-753: Surface website results on 'All results' page [BE]

### PR DESCRIPTION
**Pull request by ababic**
Ticket URL: https://national-archives.atlassian.net/browse/DF-753

## About these changes

Now that we are utilising Wagtail's built-in search mechanisms for website results, we must update the 'All results' view to show a summary of the results.

## How to check these changes

Where possible, provide guidance to help your reviewer

## Before assigning to reviewer, please make sure you have

- [ ] Checked things thoroughly before handing over to reviewer.
- [ ] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [ ] Ensured that PR includes only commits relevant to the ticket.
- [ ] Waited for all CI jobs to pass before requesting a review.
- [ ] Added/updated tests and documentation where relevant.

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: etna/search/views.py [hunk 1]
```diff
@@ -778,20 +785,40 @@ def get_buckets_for_display(self) -> Dict[str, Bucket]:
             buckets[bucket.key] = bucket
         return buckets
 
+    def process_valid_form(self, form: Form) -> HttpResponse:
+        """
+        Overrides `BaseSearchView.process_valid_form()` to query the website
+        for matches as well as the API (when a query is present).
+        """
+        if self.query:
```
[the remark this answers is not in the document]
Comment (ababic) [reply etna/search/views.py]: This has been fixed now

File: templates/search/featured_search.html [hunk 2]
```diff
@@ -31,23 +31,24 @@
...
 
-        {% if buckets.blog.result_count > 0 %}
-            {% include './blocks/featured-search__interpretive-card.html' with bucket=buckets.blog %}
-        {% endif %}
-    </div>
+    {% if website_result_count %}
+        <section class="featured-search__website-results">
+            <header class="featured-search__website-results__head">
+                <h2 class="featured-search__website-results__heading">Results from the website</h2>
+                <a href="/search/website?q={{ search_query|iriencode }}" class="featured-search__website_results-link">
```
Comment (TNA-Allan) [inline templates/search/featured_search.html:39]: For default search, the link appears as http://127.0.0.1:8000/search/website?q=None, the `search_query|iriencode` does not seem to convert None, looks like `self.query = self.form.cleaned_data.get("q", "")` is not getting its default.

**Review by TNA-Allan — CHANGES_REQUESTED** [review]

File: templates/search/featured_search.html (hunk 2 above)
Comment (ababic) [reply templates/search/featured_search.html:39]: @TNA-Allan on investigating, it looks like this is down to the `q` field on the forms having an `empty_value` value set to `None`. I'm just figuring out the best course of action here.

Comment (ababic) [reply templates/search/featured_search.html:39]: Fixed in [3d45b0d](https://github.com/nationalarchives/ds-wagtail/pull/1136/commits/3d45b0de396e9d71dfb8ec85fd9533a8fdb4dbc7)

Comment (TNA-Allan) [reply templates/search/featured_search.html:39]: Thanks @ababic

**Review by TNA-Allan — APPROVED** [review]
Looks good. Thanks @ababic

**Pull request merged**